src/sdl_utils.cpp

```diff
@@ -0,0 +1,57 @@
+#include "sdl_gl.h"
+#include <iostream>
+
+void sdl_application_abort(const char * msg)

```

Comment:
I like my function names to be camel case, starting in upper case.

In this case, I would suggest something like: AbortSDLApplication. However, you'll see in a comment below, the engine is actually handling this via destructor which is triggered from uncaught exception, (or intentionally thrown exception). And doing that makes this not really SDL related at all.

So, perhaps this would be better moved to the engine.cpp as a helper called:

AbortApplication(const char *msg, int errorCode = 1);

or something like that. Wdyt?

Reply:
You can update the Init failure cases to use that too.